edit_form.php

```diff
@@ -98,13 +98,16 @@ public function definition() {
                            get_string('onerow', 'report_customsql'));
 
         $mform->addElement('text', 'emailto', get_string('emailto', 'report_customsql'), 'size = 70');
+        $mform->addElement('text', 'customdir', get_string('customdir', 'report_customsql'), 'size = 70');
         $mform->addElement('select', 'emailwhat', get_string('emailwhat', 'report_customsql'),
                 report_customsql_email_options());
         $mform->disabledIf('singlerow', 'runable', 'eq', 'manual');
         $mform->disabledIf('at', 'runable', 'ne', 'daily');
         $mform->disabledIf('emailto', 'runable', 'eq', 'manual');
+        $mform->disabledIf('customdir', 'runable', 'eq', 'manual');
         $mform->disabledIf('emailwhat', 'runable', 'eq', 'manual');
         $mform->setType('emailto', PARAM_RAW);
+        $mform->setType('customdir', PARAM_RAW);
```

Comment:
Why is this not PARAM_PATH? Either there is a good reason, and we need a comment to explain why, and why the validation check is enough for security. Alternatively, it needs to be changed.